notebooks/406-3D-pose-estimation-webcam/406-opencv-3D-pose-estimation.ipynb

```diff
@@ -0,0 +1,587 @@
+    "    )\n",
+    "    ! $convert_command"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a4364301-3a19-4bcb-bfcf-25c53e72a818",
+   "metadata": {},
+   "source": [
+    "### Load the model\n",
+    "\n",
+    "Converted models are located in a fixed structure, which indicates vendor, model name and precision.\n",
+    "\n",
+    "Only a few lines of code are required to run the model. First, we initialize the Inference Engine. Then we read the network architecture and model weights from the .bin and .xml files to compile it for the desired device.Creates an inference request object used to infer the compiled model.The created request has allocated input and output tensors."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "92a04102-aebf-4976-874b-b98dca97ec48",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# initialize inference engine\n",
+    "ie_core = Core()\n",
+    "# read the network and corresponding weights from file\n",
+    "model = ie_core.read_model(model=ir_model_path, weights=model_weights_path)\n",
+    "# load the model on the CPU (you can use GPU or MYRIAD as well)\n",
+    "compiled_model = ie_core.compile_model(model=model, device_name=\"CPU\")\n",
+    "infer_request = compiled_model.create_infer_request()\n",
+    "input_tensor_name = model.inputs[0].get_any_name()\n",
+    "\n",
+    "# get input and output names of nodes\n",
+    "input_layer = compiled_model.input(0)\n",
+    "output_layers = list(compiled_model.outputs)\n",
+    "\n",
+    "# get input size\n",
+    "height, width = list(input_layer.shape)[2:]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e46adaa6-feae-4df0-8dba-4132ed283e02",
+   "metadata": {},
+   "source": [
+    "## Processing\n",
+    "### Model Inference\n",
+    "The image data is used as the input of the model to obtain the output heatmaps, PAF and features."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "08f8055b-a6cf-4003-8232-6f73a86d6034",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def model_infer(scaled_img, stride):\n",
+    "    \"\"\"\n",
+    "    inference\n",
+    "\n",
+    "    Parameters:\n",
+    "        scaled_img: resized image\n",
+    "        stride: int, the stride of the window\n",
+    "    \"\"\"\n",
+    "    global infer_request, input_tensor_name, model\n",
+    "\n",
+    "    n, c, h, w = model.inputs[0].shape\n",
+    "    # print([n, c, h, w ])\n",
+    "    img = scaled_img[\n",
+    "        0 : scaled_img.shape[0] - (scaled_img.shape[0] % stride),\n",
+    "        0 : scaled_img.shape[1] - (scaled_img.shape[1] % stride),\n",
+    "    ]\n",
+    "\n",
+    "    if h != img.shape[0] or w != img.shape[1]:\n",
+    "        model.reshape(\n",
+    "            {input_tensor_name: PartialShape([n, c, img.shape[0], img.shape[1]])}\n",
+    "        )\n",
+    "        compiled_model = ie_core.compile_model(model, \"GPU\")\n",
```

Comment:
Hi thanks for your clarification
1. I believe model reshaping will have impact on its accuracy, maybe this case is not included, but we do not recommend developer to change its height and width dimension;
2. in this case, since the shape of input data is static, wont be changed ？, i think we can resize the input image to fit the model's input requirements. so the 'base_height' shall be equal to model's input shape. and we dont need to create the infer_request twice.